tests/test_inngest/test_client/test_send.py

```diff
@@ -130,6 +132,98 @@ async def test_partial_send_error(self) -> None:
             )
         assert len(ctx.exception.ids) == 1
 
+    async def test_client_send_retry(self) -> None:
+        """
+        Ensure that the client retries on error when sending to event API, and
+        that retries use the same idempotency key header.
+        """
+
+        proxy_request_count = 0
+
+        # Create a proxy that mimics a request reaching the Dev Server but the
+        # client receives a 500 on the first attempt. This ensures that the Dev
+        # Server's event processing logic properly handles the idempotency key
+        # header.
+        def on_request(
+            body: typing.Optional[bytes],
+            headers: dict[str, list[str]],
+            method: str,
+            path: str,
+        ) -> http_proxy.Response:
+            nonlocal proxy_request_count
+            proxy_request_count += 1
+
+            # Always forward request to real dev server
+            resp = httpx.request(
+                method=method,
+                url=f"{dev_server.server.origin}{path}",
+                headers={k: v[0] for k, v in headers.items()},
+                content=body,
+            )
+
+            # But make client think we failed with synthetic 500 response on first try
+            if proxy_request_count == 1:
+                return http_proxy.Response(
+                    body=b"{}",
+                    headers={},
+                    status_code=500,
+                )
+            else:
+                # forward subsequent dev server responses
+                return http_proxy.Response(
+                    body=resp.content,
+                    headers=dict(resp.headers),
+                    status_code=resp.status_code,
+                )
+
+        proxy = http_proxy.Proxy(on_request)
+        proxy.start()
+        self.addCleanup(proxy.stop)
+
+        client = inngest.Inngest(
+            event_api_base_url=proxy.origin,
+            app_id="my-app",
```

Comment:
Use the `test_core.random_suffix` function for the app ID to avoid collisions with other tests